### DISCUSSION TITLE
Images are not displays at first page opening.

### DISCUSSION THREAD

**Discussion by vanyapr** [vercel/next.js#23330] [Help]
When I first open the webpage, some images don't display at all, and the browser console displays 502 errors. After page refresh everythingg is fine. This can be seen on the project https://geo-oz.ru. On a local computer with npm run dev, everything works fine, and with npm run build && npm run start there is appears such a problem. Next.js is proxied via nginx and runs via pm2. Pm2 status are displaying a bunch of next.js app crashes, but in dev mode app works without any crashes.

**Comment by akellbl4** [comment]
Hi,
Could you share what is the cause of those crashes? Any logs or smth?

**Reply by vanyapr** [reply]
> geo-oz.ru@1.1.0 start
4|geo-oz.ru        | > next start -p 8000
4|geo-oz.ru        | ready - started server on 0.0.0.0:8000, url: http://localhost:8000
4|geo-oz.ru        | info  - Using webpack 5. Reason: no next.config.js
4|geo-oz.ru        | Error: Could not find a production build in the '/user/geo-oz.ru/.next' directory. Try building your app with 'next build' before starting the production server. https://err.sh/vercel/next.js/production-start-no-build-id
4|geo-oz.ru        |     at Server.readBuildId (/user/geo-oz.ru/node_modules/next/dist/next-server/server/next-server.js:148:355)
4|geo-oz.ru        |     at new Server (/user/geo-oz.ru/node_modules/next/dist/next-server/server/next-server.js:3:120)
4|geo-oz.ru        |     at NextServer.createServer (/user/geo-oz.ru/node_modules/next/dist/server/next.js:1:2745)
4|geo-oz.ru        |     at /user/geo-oz.ru/node_modules/next/dist/server/next.js:1:3143
4|geo-oz.ru        |     at async NextServer.prepare (/user/geo-oz.ru/node_modules/next/dist/server/next.js:1:2495)
4|geo-oz.ru        |     at async /user/geo-oz.ru/node_modules/next/dist/cli/next-start.js:19:331
PM2                | App [geo-oz.ru:4] exited with code [1] via signal [SIGINT]
PM2                | App [geo-oz.ru:4] starting in -fork mode-
PM2                | App [geo-oz.ru:4] online

**Reply by vanyapr** [reply]
0 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'start' ]
1 info using npm@7.7.2
2 info using node@v12.20.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:/usr/lib/node_modules/npm/npmrc Completed in 2ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/user/geo-oz.ru/.npmrc Completed in 0ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/user/.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:/usr/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 9ms
18 timing npm:load:configload Completed in 9ms
19 timing npm:load:setTitle Completed in 0ms
20 timing npm:load:setupLog Completed in 1ms
21 timing npm:load:cleanupLog Completed in 3ms
22 timing npm:load:configScope Completed in 0ms
23 timing npm:load:projectScope Completed in 1ms
24 timing npm:load Completed in 16ms
25 timing config:load:flatten Completed in 3ms
26 timing command:run Completed in 36084ms
27 verbose stack Error: command failed
27 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
27 verbose stack     at ChildProcess.emit (events.js:314:20)
27 verbose stack     at maybeClose (internal/child_process.js:1022:16)
27 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
28 verbose pkgid geo-oz.ru@1.1.0
29 verbose cwd /user/geo-oz.ru
30 verbose Linux 4.15.0-124-generic
31 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "start"
32 verbose node v12.20.0
33 verbose npm  v7.7.2
34 error code 137
35 error path /user/geo-oz.ru
36 error command failed
37 error command sh -c next start -p 8000
38 verbose exit 137

**Reply by akellbl4** [reply]
```
4|geo-oz.ru | Error: Could not find a production build in the '/user/geo-oz.ru/.next' directory. Try building your app with 'next build' before starting the production server. https://err.sh/vercel/next.js/production-start-no-build-id
```

As I can see you don't run `next build` before starting your application.

**Reply by vanyapr** [reply]
Problem solved by increasing memory on my cloud server from 1gb to 2gb, and replacing vercel version to the latest cannary build (1.0.10-canary-9). Also i enabled http 2 on nginx.
It seems that builded next.js is requires minimum 1gb of memory, and app works incorrect without http 2.
